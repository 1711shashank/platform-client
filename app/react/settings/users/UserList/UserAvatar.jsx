import React from "react";
import PropTypes from "prop-types";

const propTypes = {
    avatar: PropTypes.string,
    realname: PropTypes.string.isRequired
};
const defaultProps = {
    avatar: "00000000000000000"
};

const UserAvatar = props => (
    <div className="listing-item-image">
        <img
            className="avatar"
            src={`https://www.gravatar.com/avatar/${props.avatar}?d=retro`}
            alt={props.realname}
        />
    </div>
);

UserAvatar.propTypes = propTypes;
UserAvatar.defaultProps = defaultProps;

export default UserAvatar;
