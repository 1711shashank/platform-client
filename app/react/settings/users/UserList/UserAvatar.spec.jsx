import React from "react";
import renderer from "react-test-renderer";
import { shallow } from "enzyme";
import UserAvatar from "./UserAvatar";

test("it renders the users avatar if it is available", () => {
    const avatar = "58dd1606bc6bd9d28f57b61cb705ff3a";
    const realname = "Test";
    const component = renderer.create(
        <UserAvatar avatar={avatar} realname={realname} />
    );
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
});
test("it renders the default gravatar-id if user-avatar is not available", () => {
    const realname = "Test";
    const component = renderer.create(<UserAvatar realname={realname} />);
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
});
test("the url for the avatar is using the user avatar prop if it is available", () => {
    const avatar = "58dd1606bc6bd9d28f57b61cb705ff3a";
    const realname = "Test";
    const userAvatar = shallow(
        <UserAvatar avatar={avatar} realname={realname} />
    );
    expect(userAvatar.find("img").prop("src")).toEqual(
        `https://www.gravatar.com/avatar/${avatar}?d=retro`
    );
});
test("the url for the avatar is using the avatar prop if it is available", () => {
    const defaultGravatar = "00000000000000000";
    const realname = "Test";
    const userAvatar = shallow(<UserAvatar realname={realname} />);
    expect(userAvatar.find("img").prop("src")).toEqual(
        `https://www.gravatar.com/avatar/${defaultGravatar}?d=retro`
    );
});

test("the realname prop is used as alt-value ", () => {
    const realname = "Test";
    const userAvatar = shallow(<UserAvatar realname={realname} />);
    expect(userAvatar.find("img").prop("alt")).toEqual(realname);
});
