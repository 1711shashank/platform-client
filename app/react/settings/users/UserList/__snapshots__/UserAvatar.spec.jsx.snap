// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`it renders the default gravatar-id if user-avatar is not available 1`] = `
<img
  alt="Test"
  className="avatar"
  src="https://www.gravatar.com/avatar/00000000000000000?d=retro"
/>
`;

exports[`it renders the users avatar if it is available 1`] = `
<img
  alt="Test"
  className="avatar"
  src="https://www.gravatar.com/avatar/58dd1606bc6bd9d28f57b61cb705ff3a?d=retro"
/>
`;
